src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -853,13 +853,17 @@ def create_application(cmd, display_name, homepage=None, identifier_uris=None,
 
 def _get_grant_permissions(graph_client, client_sp_object_id=None, query_filter=None):
     query_filter = query_filter or ("clientId eq '{}'".format(client_sp_object_id) if client_sp_object_id else None)
+    grant_info = graph_client.oauth2_permission_grant.list(filter=query_filter)
     try:
-        grant_info = graph_client.oauth2_permission_grant.list(filter=query_filter)
-    except CloudError as ex:  # Graph doesn't follow the ARM error; otherwise would be caught by msrest-azure
```

Comment:
`list` returns a `Paged` object which calls REST API **lazily**, so this `except CloudError` will never be hit.